test/inductor/test_dependencies.py

```diff
@@ -0,0 +1,64 @@
+# Owner(s): ["module: inductor"]
+import contextlib
+import unittest
+
+import torch
+
+from torch._inductor.graph import GraphLowering
+from torch._inductor.ir import Buffer, FixedLayout, Pointwise
+from torch._inductor.virtualized import ops, V
+
+from torch.testing._internal.common_utils import TestCase as TorchTestCase
+from torch.testing._internal.inductor_utils import HAS_CPU, HAS_CUDA
+
+
+class TestDependencies(TorchTestCase):
+    def _create_buffer(self, name, shape, dtype=torch.float32):
+        return Buffer(name, FixedLayout(torch.device("cuda:0"), dtype, shape))
+
+    def setUp(self):
+        super().setUp()
+
+        class DummyModule(torch.nn.Module):
+            def forward(self, x):
+                return x * 2
+
+        self._gm = torch.fx.symbolic_trace(DummyModule())
+        self._graph = GraphLowering(self._gm)
+
+        self._stack = contextlib.ExitStack()
+        self._stack.enter_context(V.set_graph_handler(self._graph))
+
+    def tearDown(self):
+        self._stack.close()
+        super().tearDown()
+
+    @unittest.skipIf(not HAS_CUDA, "CUDA-only test")
+    def test_bucketize_dependencies(self):
```

Comment:
maybe worth adding a test that tests the fusion you want to occur does occur, with `run_and_get_code`

Reply:
good idea - I added a `self.assertEqual(torch._inductor.metrics.generated_kernel_count, 1)` into a bucketize -> add test in `test_torchinductor.py`